doc/modules/clustering.rst

```diff
@@ -743,6 +750,77 @@ each class.
 
  * :ref:`sphx_glr_auto_examples_cluster_plot_agglomerative_clustering_metrics.py`
 
+Bisecting K-Means
+-----------------
+
+.. _bisect_k_means:
+
+The :class:`BisectKMeans` is an iterative variant of :class:`KMeans`, using
+divisive hierarchical clustering.
+Instead of creating all centroids at once - It obtains desired number of centroids
+by picking one cluster at each iteration and splitting it into two new clusters,
+until target number of clusters is reached.
+
+With that approach, it is possible to produce either an un-nested (flat)
+clustering or a hierarchical clustering.
+That approach is also preferable to agglomerative clustering
+if the number of clusters is small, compared to the number of data points.
+
+Cluster to split is picked depending on parameter ``bisect_strategy``:
+- Cluster with largest amount of data points (with ``largest_cluster``)
+- Cluster with biggest SSE (Sum of Squared Errors) (with ``biggest_sse``)
+
+Picking by largest amount of data points in most cases produces result as
+accurate as picking by SSE and is faster (especially for larger amount of data
+points, where calculating error may be costly).
+
+Difference between Bisect K-Means and regular K-Means can be seen on example below -
+while regular K-Means tend to create non-related clusters, clusters from Bisect K-Means
+create quite visible hierarchy.
```

Comment:
I don't understand where the visible hierarchy is. The example must be either more compelling or better explained. Adding some plot might be more intuitive, maybe the one in https://github.com/scikit-learn/scikit-learn/pull/20031#issuecomment-850650751 (see code in Details).

If the example has some plots or is more than a few lines, I would move the example in its own file `example/cluster/plot_bisect_k_means.py` (see other examples for expected syntax). Then I would link the rendered example to this guide (see other linked examples for the required syntax).

Reply:
@TomDLT I've added `examples/cluster/plot_kmeans_comparison.py`, but have problems with properly linking it. May I ask you for some tips?

Reply:
You can get the link syntax from other examples in the page.

Direct link: 
```rst
:ref:`sphx_glr_auto_examples_cluster_plot_kmeans_comparison.py`
```
Link with a picture generated by the example:
```rst
.. image:: ../auto_examples/cluster/images/sphx_glr_plot_kmeans_comparison_001.png
    :target: ../auto_examples/cluster/plot_kmeans_comparison.html
    :scale: 38
```

Reply:
However, I would rename the file to mention bisect kmeans, e.g. `plot_bisect_kmeans.py`.

Reply:
ah, ok - I did it in wrong way